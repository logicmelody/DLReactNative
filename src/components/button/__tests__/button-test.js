import React from 'react';
import Button from '../index';

import renderer from 'react-test-renderer';

const mockOpenURL = jest.fn();
const mockOnPress = jest.fn();

// 1. Set openURL module function to jest.fn
jest.mock('Linking', () => ({
	openURL: mockOpenURL,
}));

describe('Button', () => {
	describe('Rendering', () => {
		test('should match to snapshot - Primary', () => {
			const component = renderer.create(
				<Button
					label="test label"
				/>
			);

			expect(component).toMatchSnapshot();
		});

		test('should match to snapshot - Secondary', () => {
			const component = renderer.create(
				<Button
					label="test label"
					primary={false}
				/>
			);

			expect(component).toMatchSnapshot();
		});
	});

	describe('Interaction', () => {
		describe('onPressHandler', () => {
			let instance;

			beforeEach(() => {
				instance = renderer.create(
					<Button
						label="test label"
						onPress={mockOnPress}
						url="https://www.test.com"
					/>
				).getInstance();

				jest.clearAllMocks();
			});

			test('should call onPress', () => {
				instance.onPressHandler();

				expect(mockOnPress).toHaveBeenCalled();
				expect(mockOnPress).toHaveBeenCalledTimes(1);
			});

			test('should call openURL if url is provided', () => {
				instance.onPressHandler();

				expect(mockOpenURL).toHaveBeenCalled();
				expect(mockOpenURL).toHaveBeenCalledTimes(1);
				expect(mockOpenURL).toHaveBeenCalledWith('https://www.test.com');
			});

			test('should not call openURL if url is nor provided', () => {
				const innerInstance = renderer.create(
					<Button
						label="test label"
						onPress={mockOnPress}
					/>
				).getInstance();

				innerInstance.onPressHandler();

				expect(mockOpenURL).not.toHaveBeenCalled();
			});
		});
	});
});
