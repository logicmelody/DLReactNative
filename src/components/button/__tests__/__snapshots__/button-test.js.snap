// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Rendering should match to snapshot - Primary 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "#38ba7d",
      "borderBottomColor": "#1e6343",
      "borderBottomWidth": 6,
      "borderWidth": 1,
      "height": 45,
      "justifyContent": "center",
      "marginLeft": 15,
      "marginRight": 15,
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "#ffffff",
        "fontSize": 16,
        "fontWeight": "600",
      }
    }
  >
    test label
  </Text>
</View>
`;

exports[`Button Rendering should match to snapshot - Secondary 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "#f34541",
      "borderBottomColor": "#a43532",
      "borderBottomWidth": 6,
      "borderWidth": 1,
      "height": 45,
      "justifyContent": "center",
      "marginLeft": 15,
      "marginRight": 15,
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "#ffffff",
        "fontSize": 16,
        "fontWeight": "600",
      }
    }
  >
    test label
  </Text>
</View>
`;
